### ISSUE TITLE
runtime exception when pressing the "character information" button

### ISSUE THREAD

**Issue by xmcp** [nathancartlidge/powertoys-run-agda#3]
Step to reproduce:

1. Search for any symbol
2. Press the magnifier icon next to it (see the screenshot below)
3. Observe a `System.InvalidOperationException` runtime exception

![image](https://github.com/user-attachments/assets/95c2c881-956b-4340-8497-86ee1873ffaf)

The exception:

```
Version: 0.86.0.0
OS Version: Microsoft Windows NT 10.0.22631.0
IntPtr Length: 8
x64: True
Date: 24/12/10 1:30:22
Exception:
System.InvalidOperationException: Cannot start process because a file name has not been provided.
   at System.Diagnostics.Process.Start()
   at Wox.Infrastructure.Helper.OpenInShell(String path, String arguments, String workingDir, ShellRunAsType runAs, Boolean runWithHiddenWindow)
   at Wox.Infrastructure.Helper.OpenCommandInShell(String path, String pattern, String arguments, String workingDir, ShellRunAsType runAs, Boolean runWithHiddenWindow)
   at Community.PowerToys.Run.Plugin.UnicodeInput.Main.<>c__DisplayClass26_0.<LoadContextMenus>b__4(ActionContext _)
   at PowerLauncher.ViewModel.ResultViewModel.<>c__DisplayClass35_0.<LoadContextMenu>b__0(Object _)
   at System.Windows.Controls.Button.OnClick()
   at System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

The exception is due to the `Helper.OpenCommandInShell(DefaultBrowserInfo.Path, DefaultBrowserInfo.ArgumentsPattern, url)` call at [Main.cs:222](https://github.com/nathancartlidge/powertoys-run-unicode/blob/c59607673b72342a56065c2d04bdab02713e7b0a/src/Community.PowerToys.Run.Plugin.UnicodeInput/Main.cs#L222). 

It seems that `DefaultBrowserInfo.Path` is somehow empty? I'm not familiar with csharp so can't further diagnoise it. I can reliably reproduce the issue with any symbol even after restarting PowerToys.

**Comment by nathancartlidge** [comment]
Interesting! I can't reproduce this for any of the characters I've tested on my machine, but when I get a chance I'll do some digging and figure out what's going wrong between us - I'm on W11, so suspect that could be one of the reasons for the difference. I'll update this thread when I know more :)

**Comment by xmcp** [comment]
I'm also on Windows 11, so Windows 10.0.22631 is basically Windows 11 23H2.

I looked at the [implementation](https://github.com/microsoft/PowerToys/blob/95919508758e71dca88632add8a03c089a822d1c/src/modules/launcher/Wox.Plugin/Common/DefaultBrowserInfo.cs#L17) of `DefaultBrowserInfo`, and seems that we need to call `.Update()` to propagate the `.Path` field. Maybe in your setup another plugin called this method.

So if my observation is correct, we should just call `DefaultBrowserInfo.Update()` on plugin launch, and maybe add more error handling logic when the user really doesn't have a default browser.

**Comment by nathancartlidge** [comment]
Apologies - that's on me for skimming your report, didn't realise it was a version code!

Good suggestion for `DefaultBrowserInfo` - I'm now able to reproduce your issue by disabling all other modules. In the short term, while I work on releasing a fix, this issue can be mitigated by enabling the built-in "URI Handler" plugin.

From looking at the `.Update` code, it looks like that already includes error handling logic (i.e. picking Edge) if there isn't a default browser set, so just a call to `.UpdateIfTimePassed` should be sufficient

**Comment by nathancartlidge** [comment]
This issue should now be fixed in [v1.3.0](https://github.com/nathancartlidge/powertoys-run-unicode/releases/tag/v1.3.0), please re-open this thread if it hasn't worked for you!

Thank you for reporting :)
